### PR TITLE
Give the csv flexibility to onboard "not New" package to Docs.Ms

### PR DESCRIPTION
**Pull request by sima-zhu**
**Problem**:
For package which is not "New" to csv, we have to use package.json to onboard the package. However, the best practice is to depend on csv file as many as possible. The PR change here is to add flexibility to use csv to onboard a new package.

**New onboard rule:**
1. If package sets "New" to true, then we will onboard the package.
2. If package sets some value to "MSDocService", then we will onboard the package. 
3. If nothing set for these two fields, then the package will not onboard to Docs.Ms.

**Pros:**
In this way, we can also push people to update `ms.service` as many as possible. If somehow we don't have the right `ms.service`, then we can use `placeholder` for temporary placement.

Testing onboarding logic:
1. Testing in Java, removed the `durabletask-azure-functions` in [package.json](https://github.com/azure-sdk/azure-docs-sdk-java/blob/main/package.json)
2. `durabletask-azure-functions` in csv is having `functions` in `MsDocService`, and `false` in `New`.
3. After docindex pipeline, the changes show up here: https://github.com/Azure/azure-docs-sdk-java/blob/daily/2022-12-15/package.json#L1054
image

Testing on deprecated message:
1. .NET testing PR: https://github.com/Azure/azure-sdk-for-net/pull/33109
2. Testing on review site: https://review.learn.microsoft.com/en-us/dotnet/api/overview/azure/storage?view=azure-dotnet&branch=daily%2F2022-12-15-ci-succeeded
image

### CODE REVIEW

File: eng/common/scripts/Helpers/Package-Helpers.ps1 [hunk 1]
```diff
@@ -1,37 +1,47 @@
...
+        Write-Host "Found more than one package entry for $($pkg.Package) selecting the first non-hidden one."
       }
     }
-    return $packageLookup
   }
+  return $packageLookup
+}
+
+# For deprecated packages, add "(deprecated)" besides of display name.
+function GetDocsTocDisplayName($pkg) {
```
Comment (danieljurek) [inline eng/common/scripts/Helpers/Package-Helpers.ps1:40]: Where is this getting used?

Comment (sima-zhu) [reply eng/common/scripts/Helpers/Package-Helpers.ps1:40]: This is needed for language-setting in each language repo

Comment (sima-zhu) [reply eng/common/scripts/Helpers/Package-Helpers.ps1:40]: I have a PR for your reference. https://github.com/Azure/azure-sdk-for-net/pull/33109

Triggered the pipeline: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2065510&view=results

File: eng/common/scripts/Helpers/Package-Helpers.ps1 [hunk 2]
```diff
@@ -1,37 +1,47 @@
...
+  $displayName = $pkg.DisplayName
+  if ('deprecated' -eq $pkg.Support) {
+    LogWarning "The pkg $($pkg.Package) is deprecated. Add 'deprecated' besides of display name."
```
Comment (benbp) [inline eng/common/scripts/Helpers/Package-Helpers.ps1:43]:
```suggestion
    LogWarning "The pkg $($pkg.Package) is deprecated. Add 'deprecated' beside the display name."
```

**Review by benbp — APPROVED** [review]

**Pull request merged**